crawl-ref/source/dat/des/altar/overflow.des

```diff
@@ -5051,3 +5051,436 @@ MAP
+KMASK: w = no_pool_fixup
+MAP
+.......
+.P.B.P.
+.......
+ENDMAP
+
+
+NAME:   mainiacjoe_overflow_cross_hatched
+TAGS:   temple_overflow_generic_1
+NSUBST: @ = 1:@ / *:x
+SUBST:  x : xcvbwl
+SUBST:  p : cnvbtwl123
+MONS:   plant, bush, fungus
+KMASK:  wl = no_monster_gen
+KMASK:  w = no_pool_fixup
+MAP
+@xxxxx@xxxxxx
+x...........x
+x.p.ppppp.p.x
+x.p.......p.x
+x.p.p...p.p.x
+@.p.p.B.p.p.x
+x.p.p...p.p.x
+x.p.......p.x
+x.p.ppppp.p.x
+x...........x
+xxxxxxxxxxxxx
+ENDMAP
+
+
+NAME:  mainiacjoe_overflow_chevrons
+TAGS:  temple_overflow_generic_1
+SUBST: c : cvbn
+SUBST: p : cnvbtwl123G
+SUBST: q : cnvbtwl123
+MONS:  plant, bush, fungus
+KMASK: wl = no_monster_gen
+KMASK: w = no_pool_fixup
+MAP
++cccccccccccc
+c...........c
+c...........c
+c..ppppppp..c
+c..ppppppp..c
+c..pp.......c
+c..pp.......c
+c..pp..qqqq.c
+c..pp..q....c
+c..pp..q.B..c
+c......q....c
+c...........c
+ccccccccccccc
+ENDMAP
+
+NAME:  mainiacjoe_overflow_24_doors
+TAGS:  temple_overflow_generic_1
+SUBST: c : xcvbn
+MAP
+...............
+.ccc+cc+cc+ccc.
+.c.....c.....c.
+.c.....c.....c.
+.+cc+cc+cc+cc+.
+.c..c.....c..c.
+.c..c+ccc+c..c.
+.c..c.....c..c.
+.+..+..B..+..+.
+.c..c.....c..c.
+.c..c+ccc+c..c.
+.c..c.....c..c.
+.+cc+cc+cc+cc+.
+.c.....c.....c.
+.c.....c.....c.
+.ccc+cc+cc+ccc.
+...............
+ENDMAP
+
+NAME:   mainiacjoe_overflow_surrounded_room
+TAGS:   temple_overflow_generic_1
+NSUBST: X: 1:. / *:x
+SUBST:  x: x.
+NSUBST: x = 1:@ / *:x
+SUBST:  c : cnvb
+MAP
+xxxxxXXXXXXXxxxxx
+xxxxx.......xxxxx
+x.....ccccc.....X
+x.ccccc...ccccc.X
+x.+.....B.....+.X
+x.ccccc...ccccc.X
+x.....ccccc.....X
+xxxxx.......xxxxx
+xxxxxxxxxxxxxxxxx
+ENDMAP
+
+NAME:  mainiacjoe_overflow_alcove_1
+TAGS:  temple_overflow_generic_1
+SUBST: c : xcnvb
+MAP
+ccccccc
+c.....c
+c..B..c
+cc...cc
+ENDMAP
+
+NAME:  mainiacjoe_overflow_alcove_2
+TAGS:  temple_overflow_generic_1
+SUBST: c : xcnvb
+MAP
+ccccccc
+c.....c
+c..B..c
+c.....c
+c.c.c.c
+c.c.c.c
+ccc@ccc
+ENDMAP
+
+NAME:  mainiacjoe_overflow_alcove_4
+TAGS:  temple_overflow_generic_1
+SUBST: c : xcnvb
+SUBST: p : cnvbt123wl
+MONS:  plant, bush, fungus
+KMASK: wl = no_monster_gen
+KMASK: w = no_pool_fixup
+MAP
+ccccccccc
+c.......c
+c.p.B.p.c
+c.p...p.c
+c.ppppp.c
+c.......c
+ccc+++ccc
+ENDMAP
+
+NAME:    mainiacjoe_overflow_concentric_impassable
+TAGS:    temple_overflow_generic_1
+NSUBST:  x = 1:@ / *:x
+NSUBST:  l = 1:. / *:l
+NSUBST:  w = 1:. / *:w
+KMASK:   wl = no_monster_gen
+KMASK:   w = no_pool_fixup
+SHUFFLE: wl
+MAP
+xxxxxxxxxxxxx
+x...........x
+x.lllllllll.x
+x.l.......l.x
+x.l.wwwww.l.x
+x.l.w...w.l.x
+x.l.w.B.w.l.x
+x.l.w...w.l.x
+x.l.wwwww.l.x
+x.l.......l.x
+x.lllllllll.x
+x...........x
+xxxxxxxxxxxxx
+ENDMAP
+
+NAME:  mainiacjoe_overflow_catwalk
+TAGS:  temple_overflow_generic_1
+SUBST: l : lw
+KMASK: wl = no_monster_gen
+KMASK: w = no_pool_fixup
+MAP
+xxxxxxxxxxxxxxx
+xlllllllllllllx
+xlllllllllllllx
+xlllllllllllllx
+xlllB.........@
+xlllllllllllllx
+xlllllllllllllx
+xlllllllllllllx
+xxxxxxxxxxxxxxx
```

Comment:
While striding out on a catwalk is "cool", this creates cheese-able terrain for players with flight or swimming ability,

I suggest shrinking the lava slightly and adding a border of floor instead of rock so that every `l` is adjacent to some floor tile and all floor tiles are connected.